src/context/auth-context.js

```diff
@@ -0,0 +1,36 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { authReducer } from "../reducers";
+
+const initialState = {
+  user: null,
+  error: "",
+  loading: false,
+};
+
+const authContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(authReducer, initialState);
+
+  useEffect(() => {
+    const data = JSON.parse(localStorage.getItem("user"));
+    if (data) {
+      axios.defaults.headers.common["Authorization"] = `Bearer ${data.token}`;
+      dispatch({
+        type: "AUTH_SUCCESS",
+        payload: data,
+      });
+    }
```

Comment:
Wanted to know if AUTH_FAILURE is a case that needs to be handled?

Reply:
I'm not sure if it's required on initial render as data is fetched from localstorage. It's handled when user tries to login through api request